### ISSUE TITLE
creating annotated tag broke

### ISSUE THREAD

**Issue by extrawurst** [extrawurst/gitui#2126]
this is how creating annotated tags used to work (before `0.25`):

![tag-annotation](https://github.com/extrawurst/gitui/assets/776816/08d45223-6a7b-4ab6-9ffc-f031064ff02a)

now its broken. i suspect this broke during the transition to textarea in b9a2e13 by @pm100

**Comment by pm100** [comment]
to be clear 
- 't' should be a single line terminated by 'return/enter'
- 'T' should be multi line (like commit). With Enter creating a new line and ctrl-enter being confirm and close

Its funny, I thought about this only yesterday - single one prompts should  use enter not ctrl-enter to confirm and close. Same for creating a new branch. "i wonder when somebody will file a bug"

**Comment by extrawurst** [comment]
@michaelAug is on this one

**Comment by MichaelAug** [comment]
The issue is with `TextInputComponent` handling the ctrl-a keypress to move the cursor to the beginning of line.
https://github.com/extrawurst/gitui/blob/0ab1ff862592d5a16785306d64c035db24e87380/src/components/textinput.rs#L421-L435

So there's multiple ways to handle this. We could make `TextInputComponent` not handle ctrl-a keybind since there are multiple shortcuts for moving the cursor to head. Or we could change the annotation keybind to something else. Which option would you prefer @extrawurst ?

**Comment by pm100** [comment]
I would do what I did with all other 'known' keystrokes that map to things that are not hugely important to tuit xtarea.  Filter them out at the start, you will see a set of skipped keystrokes at the start of the key handling (on the road so I can't point to the specific line)

**Comment by MichaelAug** [comment]
Thanks @pm100, I think your filtering out code was removed in https://github.com/extrawurst/gitui/pull/2053 though...

**Comment by extrawurst** [comment]
@MichaelAug I had a look at #2053 again and remember what happened: the cleaner way to do it is to first react to our local events and only push down whats not handled into textarea. so I prototyped the same solution for this one: #2139

please pick it up from there:
* needs further testing
* needs to swap out the TextInput when going to `annotation mode` to a multiline one

**Comment by pm100** [comment]
@extrawurst I just saw that you removed the code that filtered out the keystrokes that have special meaning to gitui, but that are safe to not hand to tta

```

				// here all 'known' special keys for any textinput call are filtered out

				if key_match(e, self.key_config.keys.toggle_verify)
					|| key_match(e, self.key_config.keys.commit_amend)
					|| key_match(
						e,
						self.key_config.keys.open_commit_editor,
					) || key_match(
					e,
					self.key_config.keys.commit_history_next,
				) {
					return Ok(EventState::NotConsumed);
				}
```

This code in fact made sure that ctrl-a worked OK

**Comment by extrawurst** [comment]
@pm100 I am not sure what you mean, ctrl+a works:
![Screenflick Movie 27](https://github.com/extrawurst/gitui/assets/776816/ebb7b933-1f8f-4e96-be81-5eb01b57df51)

**Comment by pm100** [comment]
But ctrla should NOT work. It's has a special meaning to gityi so I filtered it out and returned as I handled. That deleted block of code picked out all the special keystrokes that have meaning to gityi text input callers. I lacked the knowledge of who the caller was so could not conditionally filter them out. However all the filtered out strokes were for relatively obscure tta features.

I had all this explained way back in the origin nal PR ( a long time ago).

I was determined to have minimal impact, so I had no idea who the caller was. I could have filtered out in the caller, or had th caller pass a list of keys that tta should not eat. But that would have been an API change
